### ISSUE TITLE
Zero ARI, NMI

### ISSUE THREAD

**Issue by Sorooshi** [abojchevski/paican#2]
Hello Aleksandar, 

Thank you for sharing your nice works with us. I have developed an algorithm and I'm trying to compare the performance of my algorithm with yours. 

As Far as I use your data in "data directory" everything works almost fine. 
The problem arises when I want to apply your algorithm on some data which are not in your data directory --let's say some synthetic data. 

Below I provide the code which I used to convert my data to a proper format for applying PAICAN.

**********************************************************************************
import numpy as np
import scipy.io as sio
import scipy.sparse as sp
from sklearn.preprocessing import OneHotEncoder

Pin = numpy.load("path_to_data/P.npy)   >>  Pin: <class 'numpy.ndarray'>, size: (200, 200)
Yin = numpy.load("path_to_data/Y.npy)   >> Yin: <class 'numpy.ndarray'>, size: (200, 5)

enc = OneHotEncoder(handle_unknown='ignore')
A = sp.csr_matrix(Pin)  # net
X = enc.fit_transform(Yin)  # features

print("X:", type(X), X.shape) >>  X: <class 'scipy.sparse.csr.csr_matrix'> (200, 17)
print("A:", type(A), A.shape) >>  A: <class 'scipy.sparse.csr.csr_matrix'> (200, 200)

************************************************************************************

Now when I apply your algorithm, ELBO values are mostly nan zeros and moreover, NMI and ARI are also zeros.

I have 160 data sets like the one which I explained above and unfortunately for all of them I obtain the same results (Zero ARI, NMI). 

Since your algorithm works well on your data set, I doubt there is something wrong with data conversion, may you please help me to figure out what's happening there?

cheers, 
-S

***********************************************************************************************

Just as the matter of completeness, below I provide the piece of code I used to apply "PAICAN":

enc = OneHotEncoder(handle_unknown='ignore')

A = sp.csr_matrix(Pin)  # net

X = enc.fit_transform(Yin)  # features

z = np.asarray(labels_true_sorted_final)
K_ = len(np.unique(z))

print(A.shape, X.shape, K_)

###### make sure the graph is undirected
A = A.maximum(A.T)

###### remove singleton nodes (without any edges)
filter_singletons = A.sum(1).A1 != 0
A = A[filter_singletons][:, filter_singletons]
X = X[filter_singletons]
z = z[filter_singletons]

###### (optionally) make sure the graph has a single connected component
cc = sp.csgraph.connected_components(A)[1]
cc_filter = cc == np.bincount(cc).argmax()

A = A[cc_filter][:, cc_filter]
X = X[cc_filter]
z = z[cc_filter]

###### Fit PAICAN
paican = PAICAN(A, X, K_, verbose=False)
z_pr, ca_pr, cx_pr = paican.fit_predict()

ari = metrics.adjusted_rand_score(z_pr, z)
nmi = metrics.normalized_mutual_info_score(z_pr, z)
print("ari:", ari, , "nmi:",  nmi)  # Always zero
ami = metrics.adjusted_mutual_info_score(z_pr, z)

**Comment by abojchevski** [comment]
Hi,

Thanks for your interest in our work.

I'm happy to have a look at if you provide the data for which you are getting the nan errors (an example `P.npy `and `Y.npy`.

**Comment by Sorooshi** [comment]
To be honest, after failing on the synthetic data, 
I tried to reproduce the result which you obtained on Lawyers data set by applying the same framework on the original Lawyers data set.

I attached the original and the converted Lawyers' data sets and the synthetic data as well.

Thanks in advance 

[data.tar.gz](https://github.com/abojchevski/paican/files/3411968/data.tar.gz)

**Comment by Sorooshi** [comment]
Ping!

**Comment by abojchevski** [comment]
I only now see that you attached the data here. I was expecting you to email it to me :)
I'll try to find some time to have a look at this in the next few days.

**Comment by Sorooshi** [comment]
I thought sending the data over there would be more convenient for later
tracks.
BTW, sorry for the inconvenience and thanks in advance.

**Comment by abojchevski** [comment]
Hi,

I'm not sure if this is still relevant, but the NAN issue is due to numerical instability with `log`, adding a small positive constant to the log is an easy way to fix this. I pushed a new version of the code with this "fix".
